### ISSUE TITLE
Add onPlaybackRateChanged (Stream<double>)

### ISSUE THREAD

**Issue by dvorapa** [luanpotter/audioplayers#385]
Similarly to onAudioPositionChanged and onDurationChanged

**Comment by dvorapa** [comment]
Might also be solved with https://github.com/luanpotter/audioplayers/issues/512

**Comment by erickzanardo** [comment]
Closing this due to reports of this being fixed on newer versions. Feel free to open a new issue in case of this still happening

**Comment by dvorapa** [comment]
I kindly disagree as there is no onPlaybackRateChanged (or similar) stream in audioplayers' documentation. This feature request haven't been solved yet.

**Comment by erickzanardo** [comment]
> I kindly disagree as there is no onPlaybackRateChanged (or similar) stream in audioplayers' documentation. This feature request haven't been solved yet. Please reopen this issue or prove it is solved

You are correct. The linked issue confused me and I though it was a merged pr instead, re opening.

**Comment by luanpotter** [comment]
Yeah this is definitely this is not implemented yet.

But I am wondering why this is needed, because playback rate is exclusively controlled by the user. duration and position change on the backend and thus need streams. We don't have a stream for volume for example.

Is there a bit more context on why this is needed?

**Comment by dvorapa** [comment]
I made it clear in #512. I finally managed to create a workaround, but it was not that easy as it might sound.

**Comment by dvorapa** [comment]
The current state is, that user has to set some rate state variable and update it every time he calls setPlaybackRate. Also he has to call a callback at all these places recalculating position and duration. So it ends up with a lot of the same code in several places. I made a workaround to do all of this stuff when position changes. This makes it more unintuitive, but works well (see https://github.com/dvorapa/stepslow/blob/master/lib/main.dart). For me this is no longer needed as my workaround works. Feel free to close it if you feel it's not worth the time spent on it.

**Comment by Gustl22** [comment]
I agree on having a variable to check the current rate, but I don't see a reason for a constantly updated stream, as it's controlled by the user.
